searx/engines/bt4g.py

```diff
@@ -0,0 +1,67 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""
+ BT4G (Videos, Music, Files)
+"""
+
+from lxml import etree
+import re
+from datetime import datetime
+from urllib.parse import quote
+from searx.utils import get_torrent_size
+
+# about
+about = {
+    "website": 'https://bt4gprx.com',
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": 'XML',
+}
+
+# engine dependent config
+categories = ['files']
+paging = True
+
+# search-url
+url = 'https://bt4gprx.com'
+search_url = url + '/search?q={search_term}&p={pageno}&page=rss'
```

Comment:
The rss feed does not support paging ...

> I did not parse the HTML page because there is an API in XML,

Not sure if this is a good decision, the RSS feed seems to be sorted "BY TIME" .. I think we need "BY RELVANCE" for a search in SearXNG ..

![image](https://github.com/searxng/searxng/assets/554536/9d8143af-8578-439f-9a2c-9b4da949d35f)

I tried some arguments on the RSS feed but I did not had luck: I didn't find a way to control the RSS result list. --> May be its better to parse the HTML output?

Reply:
Wait, seems `orderby=`  does work:

- https://bt4gprx.com/search?q=foo&orderby=time&page=rss 
- https://bt4gprx.com/search?q=foo&orderby=relevance&page=rss

Reply:
@unixfox I pushed a WIP commit on top of your branch / this commit is just to show how it works / what I have tested ..

The option `orderby=` and `category` seem to work in the RSS feed, the only drawback of the RSS feed is the fixed limit of 15 result per query term / not sure if this is good enough for a torrent search.

@Bnyro, @unixfox .. what do you think about this limited result list?

Reply:
I think that's decent if we enable paging as well. That's possible by passing the page number as `p` query parameter.

For example: https://bt4gprx.com/search?q=foo&orderby=time&page=rss&p=2

And yes, ordering by relevance is the very best choice for that engine in my opinion.

Reply:
Oh, I just saw that's what you just disabled, then I'll need to rethink my answer ...

Reply:
Pagination still seems to work when sorting by relevance, so I don't think there's any drawback if we don't sort by the time.

Reply:
> The rss feed does not support paging ...

That's not true, with the `p` query parameter it does support paging.

I'm okay with setting it to "relevant" type.

About whether using RSS or HTML, if we want to have less breakage and less burner of maintaining this engine I think we should prioritize the RSS type. It provides most of the features we need even though we miss some functionalities.

Reply:
> That's not true, with the `p` query parameter it does support paging.

Not sure where my issue is but I get always page one in my reader, even I set `p=2`.

> Pagination still seems to work when sorting by relevance

Yes, seems my feed reader is kidding me ...

> I'm okay with setting it to "relevant" type.